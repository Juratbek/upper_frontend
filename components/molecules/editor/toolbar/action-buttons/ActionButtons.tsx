import { IconMenu } from '@codexteam/icons';
import { FC } from 'react';
import { ICONS } from 'variables/icons';

import classes from './ActionButtons.module.scss';

const PlusIcon = ICONS.plus;

export const ActionButtons: FC<{ onPlus: VoidFunction; onSettings: VoidFunction }> = ({
  onPlus,
  onSettings,
}) => (
  <div className={classes.container}>
    <button className={classes.btn} onClick={onPlus}>
      <PlusIcon width={32} height={32} strokeWidth={3} />
    </button>
    <button
      className={classes.btn}
      onClick={onSettings}
      dangerouslySetInnerHTML={{ __html: IconMenu }}
    />
  </div>
);
