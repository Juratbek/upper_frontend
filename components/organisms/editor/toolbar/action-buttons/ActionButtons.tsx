import { IconMenu } from '@codexteam/icons';
import { PlusIcon } from 'components/icons';
import { forwardRef } from 'react';
import { getClassName } from 'utils';

import classes from './ActionButtons.module.scss';

export const ActionButtons = forwardRef<
  HTMLDivElement,
  {
    onPlus: VoidFunction;
    onSettings: VoidFunction;
    color: string;
  }
>(function Component({ onPlus, onSettings, color }, ref) {
  return (
    <div className={classes.container} ref={ref}>
      <button
        className={getClassName(classes.btn, 'action-button')}
        onClick={onPlus}
        aria-label='plus'
      >
        <PlusIcon color={color} width={32} height={32} />
      </button>
      <button
        aria-label='settings'
        className={getClassName(classes.btn, 'action-button')}
        onClick={onSettings}
        dangerouslySetInnerHTML={{ __html: IconMenu }}
      />
    </div>
  );
});
