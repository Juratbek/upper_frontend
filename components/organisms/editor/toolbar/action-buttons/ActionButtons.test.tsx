import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { vi } from 'vitest';

import { ActionButtons } from './ActionButtons';

describe('ActionButtons', () => {
  const mockOnPlus = vi.fn();
  const mockOnSettings = vi.fn();
  const color = 'red';

  beforeEach(() => {
    mockOnPlus.mockClear();
    mockOnSettings.mockClear();
  });

  it('renders correctly and matches snapshot', () => {
    const { container } = render(
      <ActionButtons onPlus={mockOnPlus} onSettings={mockOnSettings} color={color} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onPlus when the Plus button is clicked', async () => {
    render(<ActionButtons onPlus={mockOnPlus} onSettings={mockOnSettings} color={color} />);

    const plusButton = screen.getByRole('button', { name: /plus/i });
    expect(plusButton).toBeInTheDocument();

    userEvent.click(plusButton);

    await waitFor(() => {
      expect(mockOnPlus).toHaveBeenCalled();
    });
  });

  it('calls onSettings when the Settings button is clicked', async () => {
    render(<ActionButtons onPlus={mockOnPlus} onSettings={mockOnSettings} color={color} />);

    const settingsButton = screen.getByRole('button', { name: /settings/i });
    expect(settingsButton).toBeInTheDocument();

    userEvent.click(settingsButton);

    await waitFor(() => {
      expect(mockOnSettings).toHaveBeenCalled();
    });
  });
});
