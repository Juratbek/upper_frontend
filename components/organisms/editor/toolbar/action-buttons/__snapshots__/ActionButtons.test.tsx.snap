// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionButtons > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <button
      aria-label="plus"
      class="btn action-button"
    >
      <div
        color="red"
        height="32"
        role="icon"
        width="32"
      >
        +
      </div>
    </button>
    <button
      aria-label="settings"
      class="btn action-button"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.40999 7.29999H9.4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2.6"
        />
        <path
          d="M14.6 7.29999H14.59"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2.6"
        />
        <path
          d="M9.30999 12H9.3"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2.6"
        />
        <path
          d="M14.6 12H14.59"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2.6"
        />
        <path
          d="M9.40999 16.7H9.4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2.6"
        />
        <path
          d="M14.6 16.7H14.59"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-width="2.6"
        />
      </svg>
    </button>
  </div>
</div>
`;
